aidb/engine/approx_aggregate_engine.py

```diff
@@ -212,17 +239,21 @@ def get_additional_required_num_samples(
       self,
       query: Query,
       sample_results: List[SampledBlob],
-      estimator: Estimator
+      estimator: Estimator,
+      agg_type: exp.AggFunc,
+      agg_index: int
   ) -> int:
-
     error_target = query.error_target
     conf = query.confidence / 100.
     alpha = 1. - conf
-    pilot_estimate = estimator.estimate(sample_results, _NUM_PILOT_SAMPLES,  conf)
+    pilot_estimate = estimator.estimate(sample_results, _NUM_PILOT_SAMPLES, conf, agg_index)
 
-    agg_type = query.get_agg_type
     if agg_type == exp.Avg:
-      p_lb = statsmodels.stats.proportion.proportion_confint(len(sample_results), _NUM_PILOT_SAMPLES, alpha / 2)[0]
+      p_lb = statsmodels.stats.proportion.proportion_confint(
+        len(sample_results),
+        _NUM_PILOT_SAMPLES,
+        alpha / query.num_aggregations

```

Comment:
This doesn't look correct to me. If there is only 1 aggregation, previous code is alpha/2

Reply:
As per input parameters at [statsmodel usage](https://www.statsmodels.org/dev/generated/statsmodels.stats.proportion.proportion_confint.html)  and usage examples at [medium blog](https://akshay-a.medium.com/confidence-interval-for-population-proportion-basic-understanding-in-python-56b8cc5f8320), we can simply pass alpha and the library should use Zalpha / 2 in computing confidence interval. 

But the only doubt is that p_lb is used only in the case of avg. So, we might have to count the number of avg expressions in the query? 

However, results mostly stayed the same, and tests were passing. We can talk about it in more detail.

Reply:
1. Show me your test results, the Github check doesn't test aggregation.
2. There are several place using confidence score, why do you only change here by using Bonferroni correction?

Reply:
[Results can be found here](https://www.notion.so/AIDB-multi-aggregation-results-after-confidence-correction-0369e861b091491891cf6990372dff54?pvs=4) 

Now, I have corrected confidence everywhere it's used.

Reply:
You need to do more trials such as 100 trials for each query rather than just 1.

Reply:
Updated [results on 100 trials](https://www.notion.so/AIDB-multi-aggregation-results-after-confidence-correction-0369e861b091491891cf6990372dff54?pvs=4)

Reply:
How many failures in 100 trails?